splice_si_api/src/main/java/com/splicemachine/si/impl/store/IgnoreTxnSupplierImpl.java

```diff
@@ -87,7 +95,35 @@ private void populateIgnoreTxnCache() throws IOException {
                     }
                 }
             }
+            cache = combineOverlappingRanges(cache);
+        }
+    }
+
+    static Set<Pair<Long, Long>> combineOverlappingRanges(Set<Pair<Long, Long>> cache) {
+        BiPredicate<Pair<Long, Long>, Pair<Long, Long>> overlapping = (a, b) ->
+            (a.getFirst() < b.getFirst() && b.getFirst() < a.getSecond()) ||
```

Comment:
I think you are missing some equality checks:
a = 10, 20
b = 10, 30
should be overlapping

Reply:
Agree that 10,20 and 10,30 should be overlapping.  What about 10,30 and 30,60 ?  The comparison in the `shouldIgnore` method doesn't use equality so it wouldn't include 30 as part of the range:
`if (txnId > range.getFirst() && txnId < range.getSecond())`
Should the beginning and ending values be included as part of the range?

Reply:
Given the implementation of `shouldIgnore` I don't think [10,30] and [30,60] should be merged, but anyway, that case shouldn't happen often. I think coming back to the same transaction is more common (hence ending up with many overlapping ranges with the same start)